### PR TITLE
📖 doc: add node bootstrap troubleshooting doc

### PR DESCRIPTION
**Pull request by sbueringer**
**What this PR does / why we need it**:
Adds some guidance how node bootstrap failures can be debugged.

xref: https://github.com/kubernetes-sigs/cluster-api/issues/4583#issuecomment-838825225

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes -

### CODE REVIEW

File: docs/book/src/user/troubleshooting.md [hunk 1]
```diff
@@ -1,5 +1,23 @@
 # Troubleshooting
 
+## Node bootstrap failures
+
+Failures during Node bootstrapping can have a lot of different causes. For example, ClusterAPI resources might be
```
Comment (CecileRobertMichon) [inline docs/book/src/user/troubleshooting.md:5]:
```suggestion
Failures during Node bootstrapping can have a lot of different causes. For example, Cluster API resources might be 
```

File: docs/book/src/user/troubleshooting.md [hunk 2]
```diff
@@ -1,5 +1,23 @@
...
+1. Access the Node via ssh.
+1. Take a look at cloud-init logs via `less /var/log/cloud-init-output.log` or `journalctl -u cloud-init --since "1 day ago"`.
+   (Note: cloud-init persists logs of the commands it executes (like kubeadm) only after they have returned.)
+1. It might also be helpful to take a look at `journalctl --since "1 day ago"`.
+1. If you see that kubeadm times out waiting for the static Pods to come up, take a look at:
+   1. containerd: `crictl ps -a`, `crictl logs`, `journalctl -u containerd`
+   1. Kubelet: `journalctl -u kubelet --since "1 day ago"`
+      (Note: it might be helpful to increase the Kubelet log level by e.g. setting `--v=8` via 
+      `systemctl edit --full kubelet && systemctl restart kubelet`)
+1. If Node bootstrapping consistently fails and the kubeadm logs are not verbose enough, the `kubeadm` verbosity
```
Comment (CecileRobertMichon) [inline docs/book/src/user/troubleshooting.md:18]: this section assumes cloud-init and kubeadm which does not apply to all bootstrap providers, maybe add something like "when using [CABPK](https://github.com/kubernetes-sigs/cluster-api/tree/master/bootstrap/kubeadm)" at the top or as a separate header

**Comment by fabriziopandini** [comment]
Thanks for this write up! It will super helpful to have something where to redirect users in case of problems.
/approve

(lgtm pending comments above to be addressed)

**Comment by sbueringer** [comment]
@fabriziopandini @CecileRobertMichon ptal. I added it to the section title.

**Review by CecileRobertMichon — COMMENTED** [review]
/lgtm

**Pull request merged**